src/dhcpmon/src/dhcp_device.cpp

```diff
@@ -666,21 +700,27 @@ int dhcp_device_start_capture(dhcp_device_context_t *context,
         }
         context->snaplen = snaplen;
 
-        if (setsockopt(context->sock, SOL_SOCKET, SO_ATTACH_FILTER, &dhcp_sock_bfp, sizeof(dhcp_sock_bfp)) != 0) {
+        if (setsockopt(context->rx_sock, SOL_SOCKET, SO_ATTACH_FILTER, &dhcp_inbound_sock_bfp, sizeof(dhcp_inbound_sock_bfp)) != 0) {
             syslog(LOG_ALERT, "setsockopt: failed to attach filter with '%s'\n", strerror(errno));
```

Comment:
Same like above, maybe we just exit program if any error happened in this function.